iroha/src/sumeragi.rs

```diff
@@ -89,24 +87,24 @@ impl Sumeragi {
         if transactions.is_empty() {
             return Ok(());
         }
-        if let Role::Leader = self.role() {
+        if let Role::Leader = self.network_topology.role(&self.peer_id) {
             let block = PendingBlock::new(transactions)
                 //TODO: actually chain block?
                 .chain_first()
                 .sign(&self.public_key, &self.private_key)?;
             let minimum_quorum_of_peers = 2;
-            if self.sorted_peers.len() < minimum_quorum_of_peers {
+            if self.network_topology.sorted_peers.len() < minimum_quorum_of_peers {
```

Comment:
You can hide `minimum_quorum_of_peers` inside topology and provide method `is_quorum()` (naming may be better)